platform/ios/src/MGLScaleBar.mm

```diff
@@ -152,8 +160,42 @@ - (void)commonInit {
     [_containerView.layer addSublayer:_borderLayer];
     
     _formatter = [[MGLDistanceFormatter alloc] init];
+
+    // Image labels are now images
+    _labelImageCache              = [[NSCache alloc] init];
+    _prototypeLabel               = [[MGLScaleBarLabel alloc] init];
+    _prototypeLabel.font          = [UIFont systemFontOfSize:8 weight:UIFontWeightMedium];
+    _prototypeLabel.clipsToBounds = NO;
+
+    NSUInteger numberOfLabels = 4;
+    NSMutableArray *labelViews = [NSMutableArray arrayWithCapacity:numberOfLabels];
+
+    for (NSUInteger i = 0; i < numberOfLabels; i++) {
+        UIView *view = [[UIView alloc] init];
+        view.bounds        = CGRectZero;
+        view.clipsToBounds = NO;
+        view.contentMode   = UIViewContentModeCenter;
+        view.hidden        = YES;
+        [labelViews addObject:view];
+        [self addSubview:view];
+    }
+    _labelViews = [labelViews copy];
+
+    // Zero is a special case (no formatting)
+    [self addZeroLabel];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(currentLocaleDidChange:)
+                                                 name:NSCurrentLocaleDidChangeNotification
+                                               object:nil];
 }
 
+- (void)currentLocaleDidChange:(NSNotification*)notification {
+    // Clear the cache, so labels should be rebuilt.
+    [self.labelImageCache removeAllObjects];
```

Comment:
IIRC - no, since the code uses `NSCache` rather than a dictionary. It has behaviour to evict objects in low memory situations (though seems unpredictable).

Reply:
Right and it may not free anything if it receives a memory warning. If you consider is not necessary then I'm OK.